src/main/java/edu/ksu/canvas/attendance/services/AttendanceSectionService.java

```diff
@@ -25,9 +35,109 @@ public AttendanceSection getFirstSectionOfCourse(long canvasCourseId) {
         return sections.isEmpty() ? null : sections.get(0);
     }
 
+    public List<AttendanceSection> getSectionByCanvasCourseId(long canvasCourseId) {
+        return sectionRepository.findByCanvasCourseId(canvasCourseId);
+    }
 
     public List<AttendanceSection> getSectionsByCourse(long canvasCourseId) {
         return sectionRepository.findByCanvasCourseId(canvasCourseId);
     }
 
+    /**
+     * @throws RuntimeException when courseForm or sections list are null or empty
+     */
+    public void save(CourseConfigurationForm courseForm, long canvasCourseId) {
+        Validate.notNull(courseForm, "courseForm must not be null");
+
+        List<AttendanceSection> sections = sectionRepository.findByCanvasCourseId(canvasCourseId);
+        if(CollectionUtils.isEmpty(sections)) {
+            RuntimeException e = new RuntimeException("Cannot load data into courseForm for non-existent sections for this course");
+            throw new ContextedRuntimeException(e).addContextValue("courseId", canvasCourseId);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "courseId" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1192)